components/atoms/Select/custom-select.tsx

```diff
@@ -10,72 +13,55 @@ interface SelectProps {
-      >
-        {label && <span className="text-sm inline-flex text-light-slate-9 mr-2">{label}:</span>}
-        <input
-          value={selected.name === "" ? placeholder : selected.name}
-          readOnly
-          type="text"
-          className="overflow-scroll text-sm cursor-pointer text-light-slate-12 focus:outline-none bg-transparent"
-        />
-        <div className="w-6 h-4 relative overflow-hidden">
-          {" "}
-          <RiArrowUpSLine className="absolute bottom-1" /> <RiArrowDownSLine className="absolute top-1  " />
-        </div>
-      </div>
-      {isOpen && (
-        <div className="relative w-full">
-          <div className="left-0 right-0 transition bg-white overflow-hidden z-50 rounded-lg border shadow-superlative absolute font-normal ">
-            {options
-              ? options.map((option, index) => (
-                <div
-                  className="  cursor-pointer text-sm text-light-slate-12 hover:text-light-orange-11 hover:bg-light-orange-3 py-2 md:py-1 px-4 transition"
-                  onClick={() => handleSelected(option)}
-                  key={index}
-                >
-                  {option.name}
-                </div>
-              ))
-              : "No options"}
-          </div>
-        </div>
-      )}
+    <div className={clsx("custom-select__container relative bg-white cursor-pointer items-center overflow-x-hidden rounded-lg text-base text-light-slate-10 min-w-max", className || "")}>
+      <Select.Root onValueChange={handleSelected}>
+        <Select.Trigger
+          aria-label="Select a limit for the number of repositories to display"
+          className="custom-select__trigger inline-flex items-center text-sm text-light-slate-12 px-4 py-1.5 gap-1 w-full bg-white rounded-lg font-semibold outline-none border border-light-slate-6"
+        >
+          {label && <span className="inline-flex text-light-slate-9">{label}:</span>}
+          <Select.Value placeholder={placeholder} />
+          <Select.Icon className="w-4 h-4 relative overflow-hidden">
+            <RiArrowUpSLine className="absolute bottom-1" />
+            <RiArrowDownSLine className="absolute top-1" />
+          </Select.Icon>
+        </Select.Trigger>
+        <Select.Portal>
+          <Select.Content
+            className={clsx("absolute right-0 top-9 bg-white overflow-hidden rounded-lg border shadow-superlative", label ? "-left-17" : "left-0")}
+          >
+            <Select.Viewport>
+              {options ?
+                options.map((option) => (
+                  <Select.Item
+                    key={option.name}
+                    className="custom-select__select-item flex items-center text-sm text-light-slate-12 cursor-pointer relative px-4 py-2 md:py-1 rounded-md outline-none"
+                    value={option.name}
```

Comment:
We should be using the `value` here instead of the `name`

Reply:
Nice catch :+1: